File: mailblastr-java/src/main/java/com/mailblastr/resources/ApiKeys.java
package com.mailblastr.resources;

import com.mailblastr.ListParams;
import com.mailblastr.MailblastrResponse;
import com.mailblastr.http.ApiClient;

/**
 * API keys — {@code mailblastr.apiKeys()}. Read-only by design.
 *
 * <p>Keys are minted, re-scoped and revoked in the MailBlastr dashboard, and
 * only there: those routes accept a signed-in dashboard session, never an API
 * key. This SDK therefore deliberately exposes no method for them — a leaked
 * key cannot mint itself a replacement, widen its own permission or revoke the
 * keys around it. All it can do is read the inventory below.
 *
 * <p>{@code token} on a listed key is only the 8-character display prefix; the
 * full secret exists solely in the dashboard, at the moment the key is created.
 */
public final class ApiKeys extends Resource {
    public ApiKeys(ApiClient api) { super(api); }

    /**
     * List non-revoked keys. With no pagination params the route returns ALL
     * of them and {@code has_more:false}. {@code GET /api-keys}
     */
    public MailblastrResponse list() { return list(null); }

    public MailblastrResponse list(ListParams params) {
        return api.request("GET", "/api-keys" + paginate(params));
    }
}

File: mailblastr-python/mailblastr/api_keys.py
"""API keys resource."""

from . import http_client
from .http_client import paginate


class ApiKeys:
    """``mailblastr.ApiKeys`` — the /api-keys endpoints.

    Listing only, by design. Keys are created, re-scoped and revoked in the
    MailBlastr dashboard by a signed-in user: the API answers 403
    ``dashboard_only`` to every API-key caller on those routes, whatever the
    key's scopes. Exposing only ``list`` means a leaked key cannot mint itself
    a replacement or widen its own access."""

    @classmethod
    def list(cls, params=None):
        """List API keys (non-secret prefixes only; revoked keys excluded).
        Called with no pagination params this returns up to 1,000 keys in one
        response, with ``has_more`` set when that cap bites. GET /api-keys"""
        return http_client.request("GET", f"/api-keys{paginate(params)}")
